src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -43,7 +43,7 @@ class Crawler implements \Countable, \IteratorAggregate
     private $document;
 
     /**
-     * @var \DOMElement[]
+     * @var \DOMNode[]
```

Comment:
this change is wrong. We only ever create it with DOMElement in it, not arbitrary DOMNode

Reply:
`public function addNode(\DOMNode $node)` adds to this property.

Reply:
And is it a BC break ?
Because the following change break our test suite:
```diff
    /**
-     * @return \ArrayIterator|\DOMElement[]
+     * @return \ArrayIterator|\DOMNode[]
     */
    public function getIterator()
    {
```

```
------ --------------------------------------------------------------- 
  Line   XXXXXX/Report/ReportBuilder.php                               
 ------ --------------------------------------------------------------- 
  71     Call to an undefined method DOMNode::getAttribute().           
  78     Parameter #1 $node of class Symfony\Component\DomCrawler\Link  
         constructor expects DOMElement, DOMNode given.                 
 ------ --------------------------------------------------------------- 
```

Here is my code:
```php
        foreach ($crawler->filter('a') as $node) {
            $href = $node->getAttribute('href');

            $link = new Link($node, $crawler->getBaseHref(), 'GET');
            // ...
        }
```

Reply:
See e.g. https://github.com/symfony/symfony/blob/593ec61f605dce963de3ec0ebc1e39741e22a184/src/Symfony/Component/DomCrawler/Crawler.php#L765

The class has many similar checks.